packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -29,11 +34,13 @@ export const useAccordionHeader = (
   ref: React.Ref<HTMLElement>,
   defaultProps?: AccordionHeaderProps,
 ): AccordionHeaderState => {
-  const { headingId, panelId, onHeaderClick: onAccordionHeaderClick, open } = useAccordionItemContext();
+  const { onHeaderClick: onAccordionHeaderClick, open } = useAccordionItemContext();
   const button = useContextSelector(accordionContext, ctx => ctx.button);
   const expandIcon = useContextSelector(accordionContext, ctx => ctx.expandIcon);
   const expandIconPosition = useContextSelector(accordionContext, ctx => ctx.expandIconPosition);
   const size = useContextSelector(accordionContext, ctx => ctx.size);
+  const id = useId('accordion-header-', props.id);
+  const panel = useDescendants(accordionItemDescendantContext)[1] as AccordionItemDescendant | undefined;
```

Comment:
You could encapsulate the usages of `useDescendants` with smth like `useAccordionHeaderDescendant`, I think it adds to some nicer readability, but it's definitely a nit